pycaption/dfxp/base.py

```diff
@@ -143,38 +152,23 @@ def _find_times(self, p_tag):
         return start, end
 
     def _translate_time(self, stamp):
-        if stamp[-1].isdigit():
-            timesplit = stamp.split(':')
-
-            if len(timesplit) < 3:
-                raise CaptionReadSyntaxError(
-                    'Begin and end time should follow the '
-                    'hour:minute:seconds.milliseconds or '
-                    'hour:minute:seconds:frames format. '
-                    'Milliseconds and frames are optional.'
-                    f'Please correct the following time {stamp}')
-
-            if '.' not in timesplit[2]:
-                timesplit[2] += '.000'
-
-            secsplit = timesplit[2].split('.')
-            if len(timesplit) > 3:
-                secsplit.append(float(timesplit[3]) / 30 * 1000000)
-            while len(secsplit[1]) < 3:
-                secsplit[1] += '0'
-            microseconds = (int(timesplit[0]) * 3600000000
-                            + int(timesplit[1]) * 60000000
-                            + int(secsplit[0]) * 1000000
-                            + int(secsplit[1]) * 1000)
-            if len(secsplit) > 2:
-                microseconds += int(secsplit[2])
-
-            return microseconds
+        match = TIME_EXPRESSION_PATTERN.search(stamp)
+        if not match:
+            raise CaptionReadTimingError(
+                f'Invalid timestamp: {stamp}. Accepted formats: hh:mm:ss / '
+                'hh:mm:ss:ff / hh:mm:ss.sub-frames / time_count h|m|s|ms|f.')
+        if match.group('clock_time'):
+            microseconds = (int(match.group('hours')) * 3600000000
+                            + int(match.group('minutes')) * 60000000
+                            + int(match.group('seconds')) * 1000000)
```

Comment:
Wouldn't it be more readable to have the multiplication for converting into microseconds only done once, after the corresponding hours, minutes, and seconds were added?